### DISCUSSION TITLE
Rename "MerkleStreamer" to "MerkleLockup"

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#255] [Ideas]
Implementing [LockupTranched](https://github.com/sablier-labs/company-discussions/discussions/19) means that the `MerkleStreamer` contract will not handle just streaming models, but also discrete, delineated ones.

In light of that, It makes most sense to rename `MerkleStreamer` to `MerkleLockup`.

Side note - we've previously discussed this [here](https://github.com/sablier-labs/v2-periphery/discussions/222). But, at the time, we didn't know about LockupTranched.

**Comment by andreivladbrg** [comment]
> means that the MerkleStreamer contract will not handle just streaming models, but also discrete, delineated ones.

well, i know that in computer science, the term "streaming" is used to refer to a continuous process, not a discrete one (like how tranches would unlock money). but for our branding, can't we also refer to it as "streaming"?

if we decide not to refer to `LockupTranched` as a streaming contract, we will have to remove [this function](https://github.com/sablier-labs/v2-core/blob/a2fe652380b6eadbff17e9791a75c12cbc2e6531/src/interfaces/ISablierV2Lockup.sol#L147-L150) from the `ISablierV2Lockup` interface.

> MerkleStreamer to MerkleLockup

wouldn't using both "merkle" and "lockup" keywords be redundant, as a merkle distributor implies that the funds are frozen within it? and this is what `Lockup` means, right?

**Comment by PaulRBerg** [comment]
> for our branding, can't we also refer to it as "streaming"?

We could. Let's discuss it here: https://github.com/sablier-labs/v2-core/discussions/708

> if we decide not to refer to LockupTranched as a streaming contract, we will have to remove [this function](https://github.com/sablier-labs/v2-core/blob/a2fe652380b6eadbff17e9791a75c12cbc2e6531/src/interfaces/ISablierV2Lockup.sol#L147-L150) from the ISablierV2Lockup interface.

I gotta open a separate discussion about that. Many implications.

> wouldn't using both "merkle" and "lockup" keywords be redundant

No. Merkle is just the name of the guy who invented Merkle trees. Worldwide, very few computer scientists associate this name with crypto airdrops.

>  as a merkle distributor implies that the funds are frozen within it?

Merkle != Merkle distributor

**Comment by andreivladbrg** [comment]
> We could. Let's discuss it here

i forgot to mention that if we decide to do so (refer to unlocked tranches as streaming), then it would be totally okay to leave the contract name as it is now, `MerkleStreamer`

**Comment by PaulRBerg** [comment]
Ok. In this case, let's first settle the API of the  `LockupTranched`, and then we can return to this discussion.

**Comment by PaulRBerg** [comment]
@razgraf, do you have any thoughts on this following our call yesterday?

I'm slightly more inclined to call it `Lockup` because that would create a link between the campaign contract and the underlying distribution contract.

**Reply by razgraf** [reply]
Sure, `MerkleLockup` sounds good.

**Comment by smol-ninja** [comment]
Closing. 
- https://github.com/sablier-labs/v2-periphery/issues/261
